cmd/drand-cli/cli.go

```diff
@@ -339,13 +341,21 @@ var appCommands = []*cli.Command{
 	},
 	{
 		Name: "generate-keypair",
-		Usage: "Generate the longterm keypair (drand.private, drand.public)" +
-			"for this node.\n",
+		Usage: "Generate the longterm keypair (drand.private, drand.public) " +
+			"for this node, and load it on the drand daemon if it is up and running.\n",
 		ArgsUsage: "<address> is the address other nodes will be able to contact this node on (specified as 'private-listen' to the daemon)",
-		Flags:     toArray(folderFlag, insecureFlag, beaconIDFlag),
+		Flags:     toArray(controlFlag, folderFlag, insecureFlag, beaconIDFlag),
 		Action: func(c *cli.Context) error {
 			banner()
-			return keygenCmd(c)
+			err := keygenCmd(c)
+			if err == nil {
+				err2 := loadCmd(c)
```

Comment:
A small comment explaining why we do this here please - this is a hack so we should explain why.

Reply:
Done!